File: Admin_p/Admin_p.Web/Modules/ServerTypes/electroDB/ShoppingCartsForm.ts
import { ServiceLookupEditor, IntegerEditor, DateEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";

export interface ShoppingCartsForm {
    UserId: ServiceLookupEditor;
    ProductId: ServiceLookupEditor;
    Quantity: IntegerEditor;
    CreatedAt: DateEditor;
    UpdatedAt: DateEditor;
}

export class ShoppingCartsForm extends PrefixedContext {
    static readonly formKey = 'electroDB.ShoppingCarts';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!ShoppingCartsForm.init)  {
            ShoppingCartsForm.init = true;

            var w0 = ServiceLookupEditor;
            var w1 = IntegerEditor;
            var w2 = DateEditor;

            initFormType(ShoppingCartsForm, [
                'UserId', w0,
                'ProductId', w0,
                'Quantity', w1,
                'CreatedAt', w2,
                'UpdatedAt', w2
            ]);
        }
    }
}

File: Admin_p/Admin_p.Web/Modules/electroDB/ShoppingCarts/ShoppingCartsForm.cs
using Serenity.ComponentModel;
using System;

namespace Admin_p.electroDB.Forms;

[FormScript("electroDB.ShoppingCarts")]
[BasedOnRow(typeof(ShoppingCartsRow), CheckNames = true)]
public class ShoppingCartsForm
{
    public int UserId { get; set; }
    public int ProductId { get; set; }
    public int Quantity { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
}